src/sandstorm/indexer.capnp

```diff
@@ -0,0 +1,139 @@
+# Sandstorm - Personal Cloud Sandbox
+# Copyright (c) 2017 Sandstorm Development Group, Inc. and contributors
+# All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+@0xcae98639575b2b35;
+
+$import "/capnp/c++.capnp".namespace("sandstorm");
+
+using Grain = import "grain.capnp";
+
+interface IndexingSession(Metadata) {
+  # This is a UiView session type, created by calling UiView.newSession().
+  #
+  # Sandstorm requests a session of this type when it wants to index a grain for search purposes.
+  #
+  # Indexing sessions, like any other sessions, *must* pay attention to the UserInfo passed to
+  # `newSession()`; only content which is visible to that user can be indexed.
+  #
+  # Note that, as an optimization, Sandstorm may start out assuming that all users see exactly
+  # the same content, and may do all indexing as an anonymous user, perhaps with no permissions.
+  # However, if the app logs an ActivityEvent that specifies that it requires specific permissions
+  # or is visible only to certain users, Sandstorm uses that as a hint to index the path associated
```

Comment:
Does this mean that Sandstorm tries to index any grain that calls `SessionContext.activity()`?